model/config.go

```diff
@@ -1204,7 +1204,7 @@ func (o *Config) IsValid() *AppError {
 		return NewLocAppError("Config.IsValid", "model.config.is_valid.sql_max_conn.app_error", nil, "")
 	}
 
-	if *o.FileSettings.MaxFileSize <= 0 {
+	if o.FileSettings.MaxFileSize != nil && *o.FileSettings.MaxFileSize <= 0 {
```

Comment:
I think these need to be ORs not ANDs. The server expects these config settings to exist and will try to access them in other places as well, which will cause panics. So we should return an error if they're not set.

To make it easier on the config uploader so that they don't have to submit every single field each time (especially useful for backwards compatibility), we should add a default for each one of these that would cause a panic before in `SetDefaults()`. You can use whatever is in the `config.json` as the default

Make sense?